libfabric.spec

```diff
@@ -19,6 +19,7 @@ Group: System Environment/Libraries
 %endif
 Url: https://www.github.com/ofiwg/libfabric
 Source: https://github.com/ofiwg/%{name}/archive/v%{dl_version}.tar.gz
+Patch0: https://github.com/daos-stack/libfabric/cxi_provider.patch
```

Comment:
This link is a 404 currently and will be a circular reference even when this PR lands.  The URL here should be the upstream's commit link for this patch.

Reply:
This patch is not upstreamed anywhere. I actually modified the patch from Steve Welch to keep it compatible with v1.14.0. I should have made a note about it. That's why I added it into this repo as part of this PR, is it not good?

Reply:
Here is a link to his original patch https://github.com/ofiwg/libfabric/commit/58d5b23d34bed5b6a49ec215aa23ee5fb683e4c2 his patch actually breaks compatibility with the 1.14.0 version they have because it adds FI_ADDR_CXI after FI_ADDR_OPX, which was added at the same time as v1.15.0... so to remain compatible it needs to be just after FI_ADDR_PSMX3.

Reply:
@brianjmurrell I opened https://github.com/daos-stack/libfabric/pull/64 let me know if that's what you wanted. If yes I'll update the URL of that patch once that other PR has landed.

Reply:
As an additional note, that patch is only for the rpm, I'm not planning to have that patch applied to DAOS build so in theory I should not need to reference the patch by commit hash ???